common/src/main/java/com/wynntils/features/ui/TradeMarketPriceMatchFeature.java

```diff
@@ -0,0 +1,169 @@
+import com.wynntils.core.features.Feature;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.handlers.chat.event.ChatMessageReceivedEvent;
+import com.wynntils.mc.event.ContainerSetSlotEvent;
+import com.wynntils.models.emeralds.EmeraldModel;
+import com.wynntils.screens.base.widgets.WynntilsButton;
+import com.wynntils.utils.mc.LoreUtils;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.type.Pair;
+import com.wynntils.utils.wynn.ContainerUtils;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.gui.components.Tooltip;
+import net.minecraft.client.gui.screens.inventory.ContainerScreen;
+import net.minecraft.client.gui.screens.inventory.MenuAccess;
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.inventory.ChestMenu;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+@ConfigCategory(Category.UI)
+public class TradeMarketPriceMatchFeature extends Feature {
+    private static final StyledText CLICK_TO_SET_PRICE = StyledText.fromString("§aClick to Set Price");
+    private static final StyledText SELL_DIALOGUE_TITLE = StyledText.fromString("What would you like to sell?");
+    private static final StyledText TYPE_SELL_PRICE =
+            StyledText.fromString("§6Type the price in emeralds or type 'cancel' to cancel:");
+
+    // Test suite: https://regexr.com/7h631
+    private static final Pattern HIGHEST_BUY_PATTERN = Pattern.compile("§7Highest Buy Offer: §a(\\d+)²§8 \\(.+\\)");
+    // Test suite: https://regexr.com/7h62u
+    private static final Pattern LOWEST_SELL_PATTERN = Pattern.compile("§7Lowest Sell Offer: §a(\\d+)²§8 \\(.+\\)");
+
+    private static final int PRICE_SET_ITEM_SLOT = 12;
+    private static final int PRICE_INFO_ITEM_SLOT = 17;
+
+    private boolean sendPriceMessage = false;
+    private long priceToSend = 0;
+
+    @SubscribeEvent
+    public void onSellDialogueUpdated(ContainerSetSlotEvent.Pre e) {
+        if (!(McUtils.mc().screen instanceof ContainerScreen containerScreen)) return;
+
+        StyledText title = StyledText.fromComponent(containerScreen.getTitle());
+        if (!title.equals(SELL_DIALOGUE_TITLE)) return;
+
+        StyledText amountItemName = StyledText.fromComponent(
+                containerScreen.getMenu().getSlot(PRICE_SET_ITEM_SLOT).getItem().getHoverName());
+        if (!amountItemName.equals(CLICK_TO_SET_PRICE)) return;
+
+        removePriceButtons(containerScreen);
+
+        addPriceButtons(containerScreen);
+    }
+
+    @SubscribeEvent
+    public void onChatMessage(ChatMessageReceivedEvent e) {
+        if (!sendPriceMessage) return;
+        if (!e.getStyledText().equals(TYPE_SELL_PRICE)) return;
+
+        WynntilsMod.info("Trying to set trade market price to " + priceToSend);
+
+        McUtils.mc().getConnection().sendChat(String.valueOf(priceToSend));
+
+        sendPriceMessage = false;
```

Comment:
Can you cancel e?

Reply:
Yes, but I don't really see a reason to. It is one small message that does not block anything, and it also helps the user understand why the trade market screen closes and opens quickly. Newer users may not know that we are interacting with the chat.